autonomi/src/networking/mod.rs

```diff
@@ -0,0 +1,326 @@
+pub struct Network {
+    task_sender: Arc<tokio::sync::mpsc::Sender<NetworkTask>>,
+}
+
+impl Network {
+    /// Create a new network client
+    /// This will start the network driver in a background thread, which is a long-running task that runs until the [`Network`] is dropped
+    /// The [`Network`] is cheaply cloneable, prefer cloning over creating new instances to avoid creating multiple network drivers
+    pub fn new(initial_contacts: Vec<Multiaddr>) -> Result<Self, NoKnownPeers> {
+        let (task_sender, task_receiver) = mpsc::channel(100);
+        let mut driver = NetworkDriver::new(task_receiver);
+
+        // Bootstrap here so we can early detect a failure
+        driver.connect_to_peers(initial_contacts)?;
+
+        // run the network driver in a background task
+        tokio::spawn(async move {
+            let _ = driver.run().await;
+        });
+
+        let network = Self {
+            task_sender: Arc::new(task_sender),
+        };
+
+        Ok(network)
+    }
+
+    /// Get a record from the network
+    /// Returns the record if successful
+    /// If the record is not found, the result will be None
+    pub async fn get_record(
+        &self,
+        addr: NetworkAddress,
+        quorum: Quorum,
+    ) -> Result<Option<Record>, NetworkError> {
+        let (record, _holders) = self.get_record_and_holders(addr, quorum).await?;
+        Ok(record)
+    }
+
+    /// Get a record from the network
+    /// Returns the record if successful along with the peers that handed it to us
+    /// If the record is not found, the result will be None and an empty list of peers
+    /// If the Quorum is not met, the result will be None and the list of peers that did manage to deliver the record
+    /// As soon as the quorum is met, the request will complete and the result will be returned.
+    /// Note that the holders returned is not an exhaustive list of all holders of the record,
+    /// it only contains the peers that responded to the request before the quorum was met.
+    pub async fn get_record_and_holders(
+        &self,
+        addr: NetworkAddress,
+        quorum: Quorum,
+    ) -> Result<(Option<Record>, Vec<PeerId>), NetworkError> {
+        let (tx, rx) = oneshot::channel();
+        let task = NetworkTask::GetRecord {
+            addr,
+            quorum,
+            resp: tx,
+        };
+        self.task_sender
+            .send(task)
+            .await
+            .map_err(|_| NetworkError::NetworkDriverOffline)?;
+        rx.await?
+    }
+
+    /// Put a record to the network
+    /// When the `to` field is empty, the record is stored at the closest nodes to the record address,
+    /// else it is specifically stored to the nodes in the `to` field
+    pub async fn put_record(
+        &self,
+        record: Record,
+        to: Vec<PeerId>,
+        quorum: Quorum,
+    ) -> Result<(), NetworkError> {
+        let (tx, rx) = oneshot::channel();
+        let task = NetworkTask::PutRecord {
```

Comment:
the `ant_network::put_record` carry out ChunkProofVerification as well.
However, seems with the CLN, the `verification` step got entirely skipped.

This will create a false user experience that: a success uploaded file may failed to be downloaded, due to some of chunks failed to be uploaded, say due to quote expiration.
In that situation, `kad::put_record` still return with success, but the copy won't got stored.

This step of verification can only be skipped AFTER we swtich from `kad::put_record_to` to `request/response messaging` (which will be used in the future work of `separate payment and upload`)